Enable the use of PRAGMA Key to encrypt db

K3yr0nym0us opened the pull request:
Personally, I think that a Sqlite database that has so many features and improvements but that has no way to make it secure seems to me to be of little use, even so, I will use it with the hope that one day it will be able to encrypt its data.

m4heshd committed:
Move repetitive build commands into environment variables (#989)

* Move repetitive build commands into environment variables

* Merge 'https://github.com/mceachen/better-sqlite3/tree/dry-gha-989'

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

---------

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

m4heshd committed:
Fix the build workflow broken by #989 (#994)

* Move repetitive build commands into environment variables (#989)

* Move repetitive build commands into environment variables

* Merge 'https://github.com/mceachen/better-sqlite3/tree/dry-gha-989'

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

---------

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

* restore "env." prefixes to NODE_BUILD_CMD and ELECTRON_BUILD_CMD

---------

Co-authored-by: mceachen <mceachen@users.noreply.github.com>
Co-authored-by: Matthew McEachen <matthew@photostructure.com>

Chicken committed:
Add support for Node v20 prebuilds (#1000)

* Add support for Node v20 prebuilds

* Also test v20

m4heshd committed:
Remove node.js v14 builds and tests (runtime EOL) (#1001)

Remove node.js v14 builds and testing (runtime EOL)

arimah committed:
Fix out-of-bounds read in statement tail parser (#996)

* Fix out-of-bounds read in statement tail parser

* Add comment with link to tail parsing issue

* Fix missing increment in tail parser + tests

JoshuaWise committed:
Update SQLite to version 3.42.0 (#1011)

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

mceachen committed:
8.4.0

m4heshd committed:
Add support for electron `v25` prebuilds (#1013)

mceachen committed:
8.5.0

m4heshd committed:
Add support for electron `v26` prebuilds (#1042)

mceachen committed:
8.5.1

verhovsky committed:
prebuild for Node on macOS+arm64 and Windows+ia32 (#1002)

Prebuild for Node on other architectures as well

mceachen committed:
8.5.2

mceachen committed:
Upgrade dependencies (#1052)

Run `ncu -u`

DamienEspitallier committed:
Fix db::serialize() crashing with Electron (#1036)

Fix buffer create in sandbox mode (electron)

Commit:
Fix: Create target directory when copying   (#1030)

Fix: Allow building with space in path

cedricvanrompay committed:
fix result code type in SQliteError documentation (#1009)

Fixes https://github.com/WiseLibs/better-sqlite3/issues/1007

threema-danilo committed:
Remove SQLITE_INTROSPECTION_PRAGMAS compile-time option (#927)

From the docs:

> This option is obsolete. It used to enable some extra some extra
> PRAGMA statements such as PRAGMA function_list, PRAGMA module_list,
> and PRAGMA pragma_list, but those pragmas are now all enabled by
> default.  See SQLITE_OMIT_INTROSPECTION_PRAGMAS.

It has been obsolete since 2019-10-04 (with SQLite 3.30.0).

destyk committed:
feat(database): now `nativeBinding` supports addon object (#974)

* Added `webpack` support

* changed from `webpack_require` to `non_webpack_require`

* Update database.js

Added URL to webpack documentation

* feat(database): now `nativeBinding` supports addon object

* style(package): line break removed

---------

Co-authored-by: Matthew McEachen <matthew@photostructure.com>

JoshuaWise committed:
general cleanup

JoshuaWise committed:
upgraded to SQLite 3.43.0
added HAVE_USLEEP=1 compile-time option

JoshuaWise committed:
updated test

JoshuaWise committed:
updated test

JoshuaWise committed:
updated test

JoshuaWise committed:
updated test

JoshuaWise committed:
updated test

JoshuaWise committed:
updated test

JoshuaWise committed:
8.6.0

coltoneshaw commented:
Do we think this will be merged soon? I'm trying to debate using some other encryption packages using Better sqlite as a base or sticking with this.

JoshuaWise committed:
Update SQLite to version 3.43.1 (#1077)

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

mceachen committed:
8.7.0

Prinzhorn committed:
Update compilation.md (#1079)

m4heshd committed:
Add support for electron `v27` prebuilds (#1082)

neoxpert committed:
update prebuild and remove node-gyp version override (#1059)

m4heshd committed:
Stop building and testing against Node `v16` (runtime EOL) (#1061)

* Stop building and testing against Node `v16` (runtime EOL)

This also removes building for Electron versions that carries Node `v16`.

* Revert electron builds

JoshuaWise committed:
Update SQLite to version 3.43.2 (#1083)

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

mceachen committed:
9.0.0

fanchenio commented:
Very necessary.

mceachen commented:
Are you aware of https://github.com/m4heshd/better-sqlite3-multiple-ciphers ?

This PR must be rebased before it can be reviewed.

JoshuaWise committed:
Update SQLite to version 3.44.0 (#1097)

* Update SQLite to version 3.44.0

* see pr#1099

* pip install setuptools

* fix if

* Add Node v21 to the build matrix

---------

Co-authored-by: mceachen <mceachen@users.noreply.github.com>
Co-authored-by: Matthew McEachen <matthew@photostructure.com>

mceachen committed:
9.1.0

mceachen committed:
Fix macOS and future Alpine prebuilds (#1100)

JoshuaWise committed:
9.1.1

cclauss committed:
Upgrade GitHub actions/checkout and actions/setup-node (#1107)

* https://github.com/actions/checkout/releases
* https://github.com/actions/setup-node/releases

JoshuaWise committed:
Update SQLite to version 3.44.2 (#1112)

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

mceachen committed:
9.2.0

m4heshd committed:
Add support for electron `v28` prebuilds (#1113)

mceachen committed:
9.2.1

mceachen committed:
9.2.2

JoshuaWise committed:
Update SQLite to version 3.45.0 (#1130)

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

mceachen committed:
9.3.0

m4heshd commented:
As I remember, we had a prior conversation about this and @JoshuaWise mentioned that it would not be wise to make this library rely on another third-party extension which isn't guaranteed to be maintained. That's why I decided to create the [`better-sqlite3-multiple-ciphers`](https://github.com/m4heshd/better-sqlite3-multiple-ciphers) fork instead of adding those features to bs3 as a PR.

JoshuaWise committed:
Update SQLite to version 3.45.1 (#1133)

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

mceachen committed:
9.4.0

m4heshd committed:
Build `arm` based macOS builds natively (#1135)

m4heshd committed:
Add support for Windows `arm64` prebuilds (#1141)

mceachen committed:
9.4.1

m4heshd committed:
Add support for electron `v29` prebuilds (#1147)

mceachen committed:
9.4.2

m4heshd committed:
Temporarily disable builds for electron `v29` (#1148)

mceachen committed:
9.4.3

merceyz committed:
fix: support SpatiaLite extension (#1137)

neoxpert committed:
Support builds for Electron v29 (#1151)

* update prebuild for latest abi versions

* conditional build for Electron 29 and higher

* add Electron 29 to prebuild versions

cclauss committed:
Keep GitHub Actions up to date with GitHub's Dependabot (#1165)

Fixes warnings like at the bottom right of
https://github.com/WiseLibs/better-sqlite3/actions/runs/7997686043

* https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot
* https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem

nikwen committed:
Clarify error handling behavior for nested transactions in docs (#1160)

dependabot[bot] committed:
Bump the github-actions group with 4 updates (#1167)

Bumps the github-actions group with 4 updates: [actions/checkout](https://github.com/actions/checkout), [actions/setup-node](https://github.com/actions/setup-node), [docker/setup-qemu-action](https://github.com/docker/setup-qemu-action) and [stefanzweifel/git-auto-commit-action](https://github.com/stefanzweifel/git-auto-commit-action).


Updates `actions/checkout` from 3 to 4
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

Updates `actions/setup-node` from 3 to 4
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v3...v4)

Updates `docker/setup-qemu-action` from 1 to 3
- [Release notes](https://github.com/docker/setup-qemu-action/releases)
- [Commits](https://github.com/docker/setup-qemu-action/compare/v1...v3)

Updates `stefanzweifel/git-auto-commit-action` from 4 to 5
- [Release notes](https://github.com/stefanzweifel/git-auto-commit-action/releases)
- [Changelog](https://github.com/stefanzweifel/git-auto-commit-action/blob/master/CHANGELOG.md)
- [Commits](https://github.com/stefanzweifel/git-auto-commit-action/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: docker/setup-qemu-action
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: stefanzweifel/git-auto-commit-action
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

JoshuaWise committed:
9.4.4

JoshuaWise committed:
added flag for new python version in build jobs

JoshuaWise committed:
Merge branch 'master' of github.com:JoshuaWise/better-sqlite3

JoshuaWise committed:
fixed typo

JoshuaWise committed:
selectively add flag for macos-14

JoshuaWise committed:
9.4.5

JoshuaWise committed:
Update SQLite to version 3.45.2 (#1173)

* Update SQLite to version 3.45.2

* fix specs

---------

Co-authored-by: mceachen <mceachen@users.noreply.github.com>
Co-authored-by: Matthew McEachen <matthew@photostructure.com>

mceachen committed:
9.5.0

m4heshd committed:
Add support for electron `v30` prebuilds (#1175)

JoshuaWise committed:
Update SQLite to version 3.45.3 (#1179)

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

mceachen committed:
9.6.0

mceachen committed:
Update troubleshooting.md (#1178)

There were a bunch of windows instructions that aren't relevant anymore (Node.js on windows just works if you "install the necessary tools" now!)

mceachen committed:
Fix python setuptools on macOS (#1181)

fix macos setuptools workaround

mceachen committed:
Add Node.js v22. Drop ancient EOL versions of Node.js and Electron (#1182)

Drop unsupported versions of node and electron

mceachen committed:
10.0.0

JoshuaWise committed:
Update SQLite to version 3.46.0 (#1190)

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

mceachen committed:
10.1.0

mceachen committed:
Drop prebuilds of Node.js v21 and Electron v25 (#1191)

* Drop unsupported versions of node and electron

* Drop Node.js v21 prebuilds

* Also drop electron v25

mceachen committed:
Relax allowed timeout (GHA on macOS is slow) (#1193)

relax allowed timeout (GHA is slow)

mceachen committed:
10.1.1

mceachen committed:
11.0.0

neoxpert committed:
Support builds for Electron v31 (#1200)

* replace v8::CopyablePersistentTraits with v8::Global

* add Electron v31 to build tasks

JoshuaWise committed:
cleaned up poorly named template

mceachen committed:
Run prebuilds before publish (#1202)

mceachen committed:
11.1.0

m4heshd committed:
Fix macOS x64 builds (#1210)

LqdBcnAtWork committed:
error C2099: initializer is not a constant (#1208)

mceachen committed:
11.1.1

viceice committed:
Use node bullseye to build arm images (#1216)

mceachen committed:
11.1.2

neoxpert committed:
Support builds for Electron v32 (#1226)

* use c++20 required by v8

* adjust macro and method signatures

* replace SetAccessor with SetNativeDataProperty

* use old signatures for older node version

* install g++ 10 on ubuntu

mikeburgh committed:
Adding SQLITE_ENABLE_DBSTAT_VTAB (#1228)

mceachen committed:
11.2.0

neoxpert committed:
Prebuild for Electron 32 (#1242)

added prebuild for electron 32.0.0

mceachen committed:
11.2.1

mceachen committed:
Add new patch infra.  (#1239)

* Add new patch infra. See #1208 and #1238

* add patches to CODEOWNERS

* revert CODEOWNERS addition

JoshuaWise committed:
Update SQLite to version 3.46.1 (#1252)

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

mceachen committed:
11.3.0

GitMurf commented:
Sorry for the dumb question, but I’m a bit confused reviewing the PR history. Is this something being actively worked on with a chance for merge? Or for an encrypted SQLite database do we need to use something like https://github.com/m4heshd/better-sqlite3-multiple-ciphers?tab=readme-ov-file ?

thanks!

m4heshd commented:
@GitMurf https://github.com/WiseLibs/better-sqlite3/pull/1018#issuecomment-1920171333

GitMurf commented:
> @GitMurf https://github.com/WiseLibs/better-sqlite3/pull/1018#issuecomment-1920171333

Ok thanks @m4heshd ... I saw that but never saw a response from Josh so wasn't sure whether the conversation continued somewhere else.

So is the plan that you keep that fork up to date with better SQLite 3 and just add the encryption extension to it? Or is it forked and will have other commits / changes to it? I really love better SQLite 3 and appreciate all the work and history here so want to make sure I am able to stick to the maintainability of this repo as I trust its future :) can you explain a little bit your plan for your repo? Is it basically to just keep the same as better SQLite 3 and just have a build version that allows encryption? Are there any differences? Thanks a ton!!

oliverschwendener committed:
Prebuild for Electron 33 (#1279)

feat: add support for electron 33

mceachen committed:
11.4.0

neoxpert committed:
Added NodeJS 23 Support (#1283)

added NodeJS 23 to build and test matrix

JoshuaWise committed:
Update SQLite to version 3.47.0 (#1284)

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

mceachen committed:
11.5.0

Meztlicoatl committed:
Issue #1271: Added missing SQLite Error code identifier. (#1273)

Added new SQLite Error code identifier.

JoshuaWise committed:
ran lzz

JoshuaWise committed:
added new result codes

JoshuaWise committed:
Update SQLite to version 3.47.1 (#1296)

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

mceachen committed:
11.6.0

JoshuaWise committed:
Update SQLite to version 3.47.2 (#1303)

Co-authored-by: mceachen <mceachen@users.noreply.github.com>

mceachen committed:
11.7.0

SvetBorislavov committed:
feat: add linux/arm64 prebuilds for electron (#1306)

* add linux arm64 prebuilds for electron

Signed-off-by: Svetoslav Borislavov <svetoslav.borislavov@limechain.tech>

* Update .github/workflows/build.yml

Co-authored-by: Pavel Borisov <37436896+PavelSBorisov@users.noreply.github.com>

---------

Signed-off-by: Svetoslav Borislavov <svetoslav.borislavov@limechain.tech>
Co-authored-by: Pavel Borisov <37436896+PavelSBorisov@users.noreply.github.com>

mceachen committed:
11.7.1

m10rten commented:
@JoshuaWise We are currently searching for this feature in SQLite, would help us if this were to be available.

SvetBorislavov committed:
fix: electron prebuilds on linux arm (#1311)

Signed-off-by: Svetoslav Borislavov <svetoslav.borislavov@limechain.tech>

mceachen committed:
11.7.2

neoxpert committed:
Prebuild for Electron 34 (#1316)

add Electron 34 to prebuild versions

JoshuaWise committed:
Update SQLite to version 3.48.0 (#1315)

Co-authored-by: mceachen <216377+mceachen@users.noreply.github.com>

mceachen committed:
11.8.0

JoshuaWise committed:
added workaround for breaking change in sqlite build system for sqlite version 3.48.0
added tests for SQLITE_ENABLE_UPDATE_DELETE_LIMIT

kjackson1998 committed:
Add API doc note about statement finalization (#1310)

* Add note about statement finalization

* PR comment

alxndrsn committed:
README: add section on Upgrading (#1300)

See: https://github.com/WiseLibs/better-sqlite3/discussions/1277

Co-authored-by: alxndrsn <alxndrsn>

Prinzhorn committed:
Update README.md (#1264)

JoshuaWise committed:
updated verbiage of various documentation

JoshuaWise committed:
11.8.1

hMihaiDavid committed:
Fix small typo in api.md (#1324)

"you might except..." => "you might expect".

nikwen committed:
Update thread count calculation method in worker_threads example (#1268)

The Node.js documentation states that `os.cpus()` should not be used to calculate the amount of threads an app should launch because CPU information might be unavailable.

In Node.js v19.4.0 and v18.14.0, `os.availableParallelism()` was added for that purpose.

m4heshd committed:
Add support for electron `v35` prebuilds (#1340)

m4heshd committed:
Update the Ubuntu runner image in build script to `ubuntu-22.04` (#1341)

JoshuaWise committed:
Update SQLite to version 3.49.1 (#1346)

Co-authored-by: mceachen <216377+mceachen@users.noreply.github.com>

mceachen committed:
11.9.0

m4heshd committed:
Use the `node:18-bullseye` image for Linux x64 builds (#1349)

This fixes #1348 by building the artifacts on a base Linux image with glibc 2.31.

mceachen committed:
11.9.1

Prinzhorn committed:
Increase timeout test because of GitHub Action (#1351)

phiresky committed:
Disallow transaction to return a Promise due to unexpected behaviour (#1364)

disallow transaction to return a promise due to unexpected behaviour

jNullj committed:
docs: fix clone url at benchmark.md (#1357)

neoxpert committed:
Prebuild for Electron 36 (#1368)

add Electron 36

JoshuaWise committed:
Update SQLite to version 3.49.2 (#1370)

Co-authored-by: mceachen <216377+mceachen@users.noreply.github.com>

mceachen committed:
11.10.0

mceachen committed:
BREAKING CHANGE: drop EOL Node.js v18 and Electron v26, v27, and v28. Add node v24 to build matrix. (#1371)

BREAKING CHANGE: drop EOL Node.js v18 and Electron v26, v27, and v28 (EOL for > 1yr).

malshoff committed:
Use correct flag to determine if V8 Sandboxing is enabled (#1373)

use correct v8 sandbox flag, swap if/else

mceachen committed:
12.0.0

neoxpert committed:
Use node-abi 4.9.0 (#1385)

add override for node-abi to latest version

mceachen committed:
12.1.0

m4heshd committed:
Add support for electron `v37` prebuilds (#1386)

mceachen committed:
12.1.1

JoshuaWise committed:
Update SQLite to version 3.50.2 (#1387)

Co-authored-by: mceachen <216377+mceachen@users.noreply.github.com>

mceachen committed:
12.2.0

m4heshd committed:
Update the Windows runner on build workflow to `windows-2022` (#1388)

dependabot[bot] committed:
Bump stefanzweifel/git-auto-commit-action from 5 to 6 in the github-actions group (#1381)

Bump stefanzweifel/git-auto-commit-action in the github-actions group

Bumps the github-actions group with 1 update: [stefanzweifel/git-auto-commit-action](https://github.com/stefanzweifel/git-auto-commit-action).


Updates `stefanzweifel/git-auto-commit-action` from 5 to 6
- [Release notes](https://github.com/stefanzweifel/git-auto-commit-action/releases)
- [Changelog](https://github.com/stefanzweifel/git-auto-commit-action/blob/master/CHANGELOG.md)
- [Commits](https://github.com/stefanzweifel/git-auto-commit-action/compare/v5...v6)

---
updated-dependencies:
- dependency-name: stefanzweifel/git-auto-commit-action
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

JoshuaWise committed:
removed lzz

JoshuaWise committed:
fixed failing NODE_MODULE_VERSION check

weary-adventurer committed:
Add node:sqlite benchmark (#1334)

JoshuaWise committed:
implemented LocalVector optimizations for newer version of v8

JoshuaWise committed:
support the SQLITE_USE_URI environment variable

neoxpert committed:
Prebuild for Electron 38 (#1398)

* Update build.yml

add Electron v38

* update node-abi

update node-abi to 4.13.1

mceachen committed:
12.3.0

JoshuaWise committed:
Update SQLite to version 3.50.4 (#1403)

Co-authored-by: mceachen <216377+mceachen@users.noreply.github.com>

mceachen committed:
12.4.0

neoxpert committed:
Include all source files in package (#1405)

update file pattern in package.json

Include files from all subfolders within src.

JoshuaWise committed:
12.4.1

JoshuaWise committed:
updated macos images for github actions

JoshuaWise committed:
Update SQLite to version 3.51.0 (#1415)

Co-authored-by: mceachen <216377+mceachen@users.noreply.github.com>

neoxpert committed:
Prebuild for Electron 39 (#1414)

* update node-abi

* added switch for latest v8 api due to deprecation of context->GetIsolate

* added Electron v39 to prebuilds

* fix typo

* use GetPrototypeV2 on v8 API version 13+

JoshuaWise committed:
added GET_PROTOTYPE macro

JoshuaWise committed:
12.4.2

m4heshd committed:
Add missing macOS x64 builds (#1419)

JoshuaWise committed:
12.4.3

m4heshd committed:
Switch to `macos-15-intel` runner from `macos-15-large` (#1420)

JoshuaWise committed:
12.4.4

mceachen committed:
Add support for Node v25 and Electron 39+ prebuilds (#1422)

* chore(build): Add support for Electron 39+ builds in CI workflow by running those prebuilds with bookworm (and keeping the older prebuilds on bullseye). Fixes #1421

* chore(build): add support for node 25 (see #1412)

mceachen committed:
12.4.5

mceachen committed:
Fix npm publishing (#1423)

chore(build): add OIDC permissions and upgrade npm for trusted publishing

mceachen committed:
12.4.6

JoshuaWise committed:
Update SQLite to version 3.51.1 (#1424)

Co-authored-by: mceachen <216377+mceachen@users.noreply.github.com>

mceachen committed:
12.5.0

JoshuaWise committed:
Update SQLite to version 3.51.2 (#1436)

Co-authored-by: mceachen <216377+mceachen@users.noreply.github.com>

mceachen committed:
12.6.0

mceachen committed:
chore(build.yml): Update Electron build command to include version 40 (#1438)

* chore(build.yml): Update Electron build command to include version 40.0.0

* chore: Add EOL date for v40

mceachen committed:
12.6.1

mceachen committed:
fix build: update node-abi version in package.json to ^4.25.0 (#1439)

fix: update node-abi version in package.json to ^4.25.0

mceachen committed:
12.6.2

mceachen committed:
chore(build.yml): update Electron version support to include v41 (#1452)

m4heshd committed:
Fix Node v25 test errors (#1454)

* Fix Node v25 errors on tests

* Disable fail-fast on test jobs

JoshuaWise committed:
Update SQLite to version 3.52.0 (#1449)

Co-authored-by: mceachen <216377+mceachen@users.noreply.github.com>

mceachen committed:
Revert "Fix Node v25 test errors" (#1455)

* Revert "Fix Node v25 test errors (#1454)"

This reverts commit 128cdf41ea4530e3c472e304cbe117ab148f2bf6.

* ci: disable fail-fast in test job strategy

mceachen committed:
12.7.0

mceachen committed:
fix: use Holder() instead of This() for Electron 41 compatibility (#1456)

fix: use Holder() instead of This() in PropertyCallbackInfo contexts

PropertyCallbackInfo::This() was removed in V8 13.x (Electron 41),
causing compilation errors in NODE_GETTER functions.

JoshuaWise committed:
Update SQLite to version 3.51.2 (#1457)

Co-authored-by: mceachen <216377+mceachen@users.noreply.github.com>

mceachen committed:
12.7.1

Prinzhorn committed:
Readme: requires Node.js v20 or later (#1443)

* Requires Node.js v20 or later

* generic wording

* link

JoshuaWise committed:
Update SQLite to version 3.51.3 (#1460)

Co-authored-by: mceachen <216377+mceachen@users.noreply.github.com>

tstone-1 committed:
fix: use HolderV2() for PropertyCallbackInfo on V8 >= 12.5 (#1459)

fix: use HolderV2() for PropertyCallbackInfo on V8 >= 13

PR #1456 replaced This() with Holder() in NODE_GETTER contexts, but
Electron 41 (V8 14.6) removed both This() and Holder() from
PropertyCallbackInfo. Only HolderV2() remains (to be renamed back to
Holder() in a future V8 version per crbug.com/333672197).

Add a PROPERTY_HOLDER compat macro that uses HolderV2() on V8 >= 13
and falls back to This() on older versions. Same pattern as the
existing GET_PROTOTYPE macro.

Verified: compiles and runs on Electron 41.0.1 (V8 14.6.202).

Fixes #1458

mceachen committed:
12.8.0

JoshuaWise committed:
Update SQLite to version 3.53.0 (#1464)

Co-authored-by: m4heshd <6711514+m4heshd@users.noreply.github.com>

m4heshd committed:
12.9.0

Maxime-J committed:
Enable percentile functions (#1447)

Add SQLITE_ENABLE_PERCENTILE compile-time option

m4heshd committed:
Add support for electron v42 prebuilds (#1466)

m4heshd committed:
12.9.1

JoshuaWise committed:
Update SQLite to version 3.53.1 (#1467)

Co-authored-by: m4heshd <6711514+m4heshd@users.noreply.github.com>

m4heshd committed:
Add support for Node.js v26 prebuilds and remove EOL builds (#1468)

* Add support for Node.js v26 prebuilds

* Update `mocha` to the latest version

* Add Node v26 to supported engines

m4heshd committed:
Temporarily rollback support for Electron v42 prebuilds (#1470)

Temporarily rollback support for electron v42 prebuilds

Avoid building for electron v42 till the updated V8 API errors are sorted out.

m4heshd committed:
12.10.0

JoshuaWise committed:
Update SQLite to version 3.53.2 (#1483)

Co-authored-by: m4heshd <6711514+m4heshd@users.noreply.github.com>

dependabot[bot] committed:
Bump the github-actions group across 1 directory with 4 updates (#1451)

Bumps the github-actions group with 4 updates in the / directory: [actions/checkout](https://github.com/actions/checkout), [actions/setup-node](https://github.com/actions/setup-node), [docker/setup-qemu-action](https://github.com/docker/setup-qemu-action) and [stefanzweifel/git-auto-commit-action](https://github.com/stefanzweifel/git-auto-commit-action).


Updates `actions/checkout` from 4 to 6
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v6)

Updates `actions/setup-node` from 4 to 6
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v4...v6)

Updates `docker/setup-qemu-action` from 3 to 4
- [Release notes](https://github.com/docker/setup-qemu-action/releases)
- [Commits](https://github.com/docker/setup-qemu-action/compare/v3...v4)

Updates `stefanzweifel/git-auto-commit-action` from 6 to 7
- [Release notes](https://github.com/stefanzweifel/git-auto-commit-action/releases)
- [Changelog](https://github.com/stefanzweifel/git-auto-commit-action/blob/master/CHANGELOG.md)
- [Commits](https://github.com/stefanzweifel/git-auto-commit-action/compare/v6...v7)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: actions/setup-node
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: docker/setup-qemu-action
  dependency-version: '4'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: stefanzweifel/git-auto-commit-action
  dependency-version: '7'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tstone-1 committed:
Fix V8 external API usage for Electron 42 (#1475)

Electron 42 ships V8 14.8, where v8::External::New and v8::External::Value require an ExternalPointerTypeTag. This breaks native rebuilds with Electron 42 headers.

Add small compatibility macros for v8::External creation/access, following the existing V8 version-guard pattern used for prior Electron compatibility fixes such as #1459. Also pass nullptr to SetNativeDataProperty to avoid ambiguous overload resolution for a missing setter.

Verified with npm test and an Electron 42.1.0 rebuild from source.

JoshuaWise committed:
12.10.1

m4heshd committed:
Add missing Electron v42 build target (#1486)

m4heshd committed:
12.11.0

m4heshd committed:
Fix Electron v42 build errors on Windows (#1488)

m4heshd committed:
12.11.1

m4heshd committed:
Add support for electron v43 prebuilds (#1490)

m4heshd committed:
12.11.2

JoshuaWise committed:
Update SQLite to version 3.53.3 (#1492)

Co-authored-by: m4heshd <6711514+m4heshd@users.noreply.github.com>

neoxpert committed:
Utilize node:20-trixie for Electron v43 and onwards (#1495)

* Update build.yml

Extended build workflow for Electron v43 and onwards as some macros from the v8 headers are not correctly expanded with GCC 12 and GCC 13 / 14 is only available since Debian Trixie.

* adjusted ENV variable naming

m4heshd committed:
12.12.0

JoshuaWise committed:
Refactored codebase to use node-addon-api (#1498)

* refactored codebase to use node-addon-api and publish with prebuilt binaries

* upgraded to sqlite 3.53.3

* fixed typo

* fixed test for windows

* mitigated performace regression from NAPI refactor

* mitigated performace regression from NAPI refactor

* mitigated performace regression from NAPI refactor

* mitigated performace regression from NAPI refactor

* fixed performance regression to surpass node:sqlite

* improved array optimization for large arrays

* further improved performance

* removed dead code

dennismutuku2005 committed:
Fix `SqliteError` cross-realm and `Error.isError` compatibility (#1473)

* Refactor SqliteError to use class syntax

* Rewrite `SqliteError` as an ES6 `class extends Error`. Fixes #1472

JoshuaWise committed:
fixed tests for SqliteError

JoshuaWise committed:
updated github workflow dep versions

JoshuaWise committed:
implemented Statement#toString()

JoshuaWise committed:
added docs for Statement#toString()

JoshuaWise committed:
implemented Database#explain()

JoshuaWise committed:
fixed typos

JoshuaWise committed:
13.0.0

JoshuaWise committed:
fixed overly strict regression in parameter binding

JoshuaWise committed:
Merge remote-tracking branch 'origin/master'

JoshuaWise committed:
13.0.1

JoshuaWise committed:
updated readme

JoshuaWise committed:
fixed typo in docs

neoxpert committed:
add gypfile option to package.json (#1505)

JoshuaWise committed:
Update SQLite to version 3.53.4 (#1506)

Co-authored-by: m4heshd <6711514+m4heshd@users.noreply.github.com>

theosotr committed:
validate the parameters of table properly (#1504)

JoshuaWise committed:
fixed segfault from making calls while worker thread is terminating

anupamme committed:
fix: this dependabot configuration does not set a co... in... (#1508)

fix: package_managers.dependabot.dependabot-missing-cooldown.dependabot-missing-cooldown security vulnerability

Automated security fix generated by OrbisAI Security

JoshuaWise committed:
13.0.2

neoxpert committed:
use Ubuntu 22.04-arm (#1510)

JoshuaWise committed:
13.0.3
